\documentclass{subfiles}
\begin{document}
    Multiple alignment is based on the phylogenetic trees. These are 
        trees in which the nodes represent the taxonomic units, while the branches 
        describe the relashionship between them. Each internal node has three branches:
        one that leads to the ancestor and two to the descendants.

    These kind of trees can by distinguished by the meaning of the branches lengths:
        if a meaning is assigned to the length we talk about \emph{cladograms},
        otherwise we talk about \emph{phylogram}. In turns we can make an additional 
        distinction: rooted and unrooed trees, whose meaning is pretty obvious. 
        The former give qualitative informations about the species, the latter quantitative ones.

    We consider two methods to build evolutionary trees: UPGMA and Neighbor-Join. 
    Alternatives to the one proposed do exist, but make use of charcter state-base optimizations,
    which is far from our interests.

    \subsubsection{UPGMA}
    \subfile{../subsub/4.2.1 - UPGMA}

    \subsubsection{Neighbor-Join}
    \subfile{../subsub/4.2.2 - Neighbor-join}
\end{document}
